homeassistant/components/advantage_air/climate.py

```diff
@@ -0,0 +1,278 @@
+    """AdvantageAir Climate class."""
+
+    def __init__(self, instance):
+        """Initialize the base Advantage Air climate entity."""
+        super().__init__(instance["coordinator"])
+        self.async_change = instance["async_change"]
+        self.device = instance["device"]
+
+    @property
+    def temperature_unit(self):
+        """Return the temperature unit."""
+        return TEMP_CELSIUS
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported temperature step."""
+        return PRECISION_WHOLE
+
+    @property
+    def max_temp(self):
+        """Return the maximum supported temperature."""
+        return 32
+
+    @property
+    def min_temp(self):
+        """Return the minimum supported temperature."""
+        return 16
+
+    @property
+    def device_info(self):
+        """Return parent device information."""
+        return self.device
+
+
+class AdvantageAirAC(AdvantageAirClimateEntity):
+    """AdvantageAir AC unit."""
+
+    def __init__(self, instance, ac_key):
+        """Initialize the Advantage Air AC climate entity."""
+        super().__init__(instance)
+        self.ac_key = ac_key
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self.coordinator.data["aircons"][self.ac_key]["info"]["name"]
+
+    @property
+    def unique_id(self):
+        """Return a unique id."""
+        return f'{self.coordinator.data["system"]["rid"]}-{self.ac_key}'
+
+    @property
+    def target_temperature(self):
+        """Return the current target temperature."""
+        return self.coordinator.data["aircons"][self.ac_key]["info"]["setTemp"]
+
+    @property
+    def hvac_mode(self):
+        """Return the current HVAC modes."""
+        if self.coordinator.data["aircons"][self.ac_key]["info"]["state"] == STATE_ON:
+            return ADVANTAGE_AIR_HVAC_MODES.get(
+                self.coordinator.data["aircons"][self.ac_key]["info"]["mode"]
+            )
+        return HVAC_MODE_OFF
+
+    @property
+    def hvac_modes(self):
+        """Return the supported HVAC modes."""
+        return [
+            HVAC_MODE_OFF,
+            HVAC_MODE_COOL,
+            HVAC_MODE_HEAT,
+            HVAC_MODE_FAN_ONLY,
+            HVAC_MODE_DRY,
+        ]
+
+    @property
+    def fan_mode(self):
+        """Return the current fan modes."""
+        return ADVANTAGE_AIR_FAN_MODES.get(
+            self.coordinator.data["aircons"][self.ac_key]["info"]["fan"], FAN_OFF
+        )
+
+    @property
+    def fan_modes(self):
+        """Return the supported fan modes."""
+        return [FAN_AUTO, FAN_LOW, FAN_MEDIUM, FAN_HIGH]
+
+    @property
+    def supported_features(self):
+        """Return the supported features."""
+        return SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+
+    @property
+    def device_state_attributes(self):
+        """Return additional attributes about AC unit."""
+        return self.coordinator.data["aircons"][self.ac_key]["info"]
```

Comment:
Please set the attributes explicitly so we can see the keys and their formatting.

Reply:
Removed the additional attributes instead. If any are user relevant I will create new entities in the sensor platform.